lib/ast.js

```diff
@@ -120,6 +120,13 @@ var AST_Node = DEFNODE("Node", "start end", {
             ctor.prototype._validate.call(this);
         } while (ctor = ctor.BASE);
     },
+    validate_ast: function() {
+        var marker = {};
+        this.walk(new TreeWalker(function(node) {
+            if (node.validate_visited === marker) throw new Error("invalid node reuse: " + node);
```

Comment:
`+ node` results in: `ERROR: invalid node reuse: [object Object]` - not useful debug info.

`+ node.print_to_string()` is what you want.

Same issue with:
```
lib/ast.js:                throw new Error("invalid transformed value: " + node);
```

Reply:
Oops, I just copied the code from that line you quoted without thinking.

We may not get a successful `AST_Node.print_to_string()`, so we'll probably need a try-catch then fall back to may be `AST_Node.TYPE` or something.

Reply:
Or just emit `AST_Node.TYPE` and `file:line:col`.